Examples/Algorithms/TrackFinding/src/TrackFindingAlgorithm.cpp

```diff
@@ -357,15 +352,33 @@ ProcessCode TrackFindingAlgorithm::execute(const AlgorithmContext& ctx) const {
 
   // Set the CombinatorialKalmanFilter options
   TrackFinderOptions firstOptions(ctx.geoContext, ctx.magFieldContext,
-                                  ctx.calibContext, slAccessorDelegate,
-                                  extensions, firstPropOptions);
+                                  ctx.calibContext, extensions,
+                                  firstPropOptions);
+
+  MeasurementSelector measSel{
+      Acts::MeasurementSelector(m_cfg.measurementSelectorCfg)};
+
+  using TrackStateCandidateCreatorType =
+      Acts::TrackStateCandidateCreatorImpl<IndexSourceLinkAccessor::Iterator,
+                                           TrackContainer>;
+  TrackStateCandidateCreatorType trackStateCreator{
+      slAccessorDelegate,
+      Acts::makeDelegate<&MeasurementCalibratorAdapter::calibrate>(calibrator),
+      Acts::makeDelegate<&MeasurementSelector::select>(measSel)};
```

Comment:
You create a not properly initialized object, and the compiler would not tell You if You missed something.

Reply:
Since these are supposed to be runtime configurable, at least optionally, I don't see this as an issue. What we do everywhere else is either introduce default functions (which you do) than can error on invocation, or when pushing the object to a consumer it can check if all of the delegates are connected.

Reply:
I don't have a strong opinion. But if it is preferred I can remove these helper methods and the starting point would be an uninitialized object and it is up to the user to remember what needs to be connected.

Though I don't see it as an improvement to move something that could be caught during compile time to the runtime. But of course You are right that delegates could be fully configurable thus it often is a runtime problem anyways.

It also seems to simplify client code. There are a few cases like 
```
Delgate delegate;
delegate.connect<&func>(obj):
Options option(delegate);
```
which just becomes 
```
Options option(makeDelegate<&func>(obj));
```
Of course the first version could just be 
```
Options option;
option.delegate.connect<&func>(obj):
```
But then the delegate has to get a default value also in cases in which there is no meaningful default value, and in which the default value is only short lived.

Reply:
I would prefer this, yes. There's also the added benefit that explicitly writing the delegate being connected avoids any ordering confusion (although this is a compile error due to mismatching types).

Reply:
I don't understand the argument about ordering confusion, either way You cannot make a "connected" delegate to a non existing object or function by accident, or at least I do not see how one snippet is more likely for errors than the other, except for uninitialized member delegates which are caught at runtime in one case and at compile time in the other. 

But I see You prefer for some reason unused default values and or more boiler plate code.